components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/ApplicationDataRetrievalClient.java

```diff
@@ -96,6 +98,124 @@ public String getApplicationAccessURL(String tenant, String applicationName)
         return StringUtils.EMPTY;
     }
 
+    /**
+     * Gets the access url configured for the given application.
+     *
+     * @param tenant tenant domain of the application
+     * @param applicationId UUID of the application
+     * @return the access url configured for the given application
+     * @throws ApplicationDataRetrievalClientException if IO exception occurs or access URL is not configured
+     */
+    public String getApplicationAccessURLForUUID(String tenant, String applicationId)
+            throws ApplicationDataRetrievalClientException {
+
+        try (CloseableHttpClient httpclient = HttpClientBuilder.create().useSystemProperties().build()) {
```

Comment:
Cannot we extract this client logic out? let's try to reduce the code duplication.

Reply:
Addressed with 2be0579cdf1d3548b7c3755834647f3a27153d2e